js/types.d.ts

```diff
@@ -2,10 +2,10 @@
 export type TypedArray = Uint8Array | Float32Array | Int32Array;
 
 export interface ModuleInfo {
-  moduleName?: string;
-  filename?: string;
-  sourceCode?: string;
-  outputCode?: string;
+  moduleName: string | null;
+  filename: string | null;
+  sourceCode: string | null;
+  outputCode: string | null;
```

Comment:
Is this not the same thing?

Reply:
I highly recommend you to watch this: https://youtu.be/f6TCB61fDwY?t=655

Reply:
Under strict mode, `undefined` and `null` are incompatible.  Because when we create the object literal, we are using functions that return `null` but _always_ defining the properties.  We could retain that optional properties, but if we use `null` for a not present value, we have to include it as a potential type.

Reply:
ok - makes sense